src/dotnet-test-nunit/TestRunner.cs

```diff
@@ -236,6 +236,12 @@ void SetupLabelOutput(TestExecutionListener listener)
                  ? _options.DisplayTestLabels.ToUpperInvariant()
                  : "ON";
 
+            listener.TestStarted += (sender, args) =>
+            {
+                if (labels == "ALL")
+                    WriteLabelLine(args.TestName);
+            };
+
             listener.TestFinished += (sender, args) =>
             {
                 if (labels == "ALL")
```

Comment:
If we are going to add the labels ALL to the start, let's remove it from here. When we go multi-threaded, the currentLabel code you pointed to could fail to prevent the second print.